packages/twenty-front/src/pages/settings/roles/components/RoleAssignment.tsx

```diff
@@ -1,19 +1,230 @@
+  });
+
+  const [modalMode, setModalMode] = useState<ModalMode | null>(null);
+  const [selectedWorkspaceMember, setSelectedWorkspaceMember] =
+    useState<SelectedWorkspaceMember | null>(null);
+  const { data: rolesData } = useGetRolesQuery();
+  const { closeDropdown } = useDropdown('role-member-select');
+  const [searchFilter, setSearchFilter] = useState('');
+
+  const workspaceMemberRoleMap = useMemo(() => {
+    if (!rolesData?.getRoles) return new Map();
+
+    const roleMap = new Map<string, { id: string; label: string }>();
+    rolesData.getRoles.forEach((role) => {
+      role.workspaceMembers.forEach((member) => {
+        roleMap.set(member.id, { id: role.id, label: role.label });
+      });
+    });
+    return roleMap;
+  }, [rolesData?.getRoles]);
+
+  const filteredWorkspaceMembers = useMemo(() => {
+    if (!searchFilter) return role.workspaceMembers;
+
+    const searchTerm = searchFilter.toLowerCase();
+    return role.workspaceMembers.filter((member) => {
+      const firstName = member.name.firstName?.toLowerCase() || '';
+      const lastName = member.name.lastName?.toLowerCase() || '';
+      const email = member.userEmail?.toLowerCase() || '';
+
+      return (
+        firstName.includes(searchTerm) ||
+        lastName.includes(searchTerm) ||
+        email.includes(searchTerm)
+      );
+    });
+  }, [role.workspaceMembers, searchFilter]);
+
+  const handleModalClose = () => {
+    setModalMode(null);
+    setSelectedWorkspaceMember(null);
+  };
+
+  const handleSelectWorkspaceMember = (workspaceMember: WorkspaceMember) => {
+    const existingRole = workspaceMemberRoleMap.get(workspaceMember.id);
+
+    setSelectedWorkspaceMember({
+      id: workspaceMember.id,
+      name: `${workspaceMember.name.firstName} ${workspaceMember.name.lastName}`,
+      role: existingRole,
+    });
+    setModalMode('assign');
+    closeDropdown();
+  };
+
+  const handleRemoveClick = (workspaceMember: WorkspaceMember) => {
+    setSelectedWorkspaceMember({
+      id: workspaceMember.id,
+      name: `${workspaceMember.name.firstName} ${workspaceMember.name.lastName}`,
+      role: workspaceMemberRoleMap.get(workspaceMember.id),
+    });
+    setModalMode('remove');
+  };
+
+  const handleConfirm = async () => {
+    if (!selectedWorkspaceMember || !modalMode) return;
+
+    await updateWorkspaceMemberRole({
+      variables: {
+        workspaceMemberId: selectedWorkspaceMember.id,
+        roleId: modalMode === 'assign' ? role.id : null,
+      },
+    });
+
+    handleModalClose();
+  };
+
+  const handleRoleClick = (roleId: string) => {
+    navigateSettings(SettingsPath.RoleDetail, { roleId });
+    handleModalClose();
+  };
+
+  const handleSearchChange = (text: string) => {
+    setSearchFilter(text);
+  };
+
   return (
-    <Section>
-      <H2Title
-        title={t`Assigned members`}
-        description={t`This Role is assigned to these workspace member.`}
-      />
-    </Section>
+    <>
+      <Section>
```

Comment:
this whole component looks a bit specific and will likely lead to duplicated code, first time we have a dropdown search in the Settings?
would be great to make it generic in upcoming work if we want to have a WorkspaceMember picker